src/explore-education-statistics-admin/src/pages/release/data/utils/__tests__/getApiDataSetLocationCodes.test.ts

```diff
@@ -0,0 +1,49 @@
+import getApiDataSetLocationCodes from '@admin/pages/release/data/utils/getApiDataSetLocationCodes';
+
+describe('getApiDataSetLocationCodes', () => {
+  test('returns the correct codes when code and oldCode are present', () => {
+    expect(
+      getApiDataSetLocationCodes({
+        label: 'Test location',
+        code: 'test-code',
+        oldCode: 'test-old-code',
+      }),
+    ).toEqual('Code: test-code, old code: test-old-code');
+  });
+
+  test('returns the correct codes when urn and laEstab are present', () => {
+    expect(
+      getApiDataSetLocationCodes({
+        label: 'Test location',
+        urn: 'test-urn',
+        laEstab: 'test-laEstab',
+      }),
+    ).toEqual('URN: test-urn, LAESTAB: test-laEstab');
+  });
+
+  test('returns the correct code when only ukprn is present', () => {
+    expect(
+      getApiDataSetLocationCodes({
+        label: 'Test location',
+        ukprn: 'test-ukprn',
+      }),
+    ).toEqual('test-ukprn');
+  });
+
+  test('returns the correct code when only code is present', () => {
+    expect(
+      getApiDataSetLocationCodes({
+        label: 'Test location',
+        code: 'test-code',
+      }),
+    ).toEqual('test-code');
+  });
+
+  test('returns undefined when no codes are present', () => {
+    expect(
+      getApiDataSetLocationCodes({
+        label: 'Test location',
+      }),
+    ).toEqual(undefined);
+  });
```

Comment:
As per suggested modifications to this, we should write a few more test cases for:

- Codes with empty strings
- Unexpected combinations of codes in the candidate e.g. `code`, `ukprn`, `laEstab`

We should always end up with a nicely formatted string that outputs as many of the codes as possible (for debugging).